pkg/metrics/prometheus.go

```diff
@@ -0,0 +1,152 @@
+package metrics
+
+import (
+	"github.com/grafana/grafana/pkg/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"net/http"
+	"strings"
+)
+
+const (
+	namespace = "grafana"
+	subsystem = "backend"
+)
+
+type PrometheusPublisher struct {
+	CounterCollectors map[string]*prometheus.CounterVec
+	GaugeCollectors   map[string]*prometheus.GaugeVec
+}
+
+func CreatePrometheusPublisher() (*PrometheusPublisher, error) {
+	go func() {
+		http.Handle("/metrics", prometheus.Handler())
+		err := http.ListenAndServe(":9200", nil)
```

Comment:
> It doesn't, but if you want to support multiple different monitoring systems it's a good choice.

Great so how do we do it without using the lib. The internal metrics lib already supports multiple publishers. 

> I wouldn't agree there. It's similar complexity math both ways.
I do not agree with you there. Dealing with raw counters and roll ups is very in accurate. Can not getting accurate sums over time. At least not found any way to do with existing TSDS (unless you store counter deltas). I mean theoretically it just math, but actually working with existing systems and having the same query to query against recent high res data and the same query against a long time range (targeting roll up periods). But might be wrong. 

> That's not particularly useful data. Those look like Dropwizard metrics
Dropwizard?? Could you describe in what way they are not useful? They are missing sum you say, sum of all the timing measurements? How would that be useful? Maybe if we also had histogram buckets, but we don't so not seeing your point. Sorry very lost :) 

But maybe better include histogram buckets, instead of pre calculated latencies as buckets can be aggregated if you have multiple grafana instances. 

How do prometheus exporter consume timings from other systems? I would love to support prometheus out of the box but maybe better to have a node exporter plugin so it can scrape the metrics endpoint and process it however prometheus wants it :)